kratos/containers/geometry_container.h

```diff
@@ -1,374 +1,374 @@
+
+    /// Const Geometry Iterator
+    typedef typename GeometriesContainerType::const_iterator GeometryConstantIterator;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default Constructor
+    GeometryContainer()
+        : mGeometries()
+    {}
+
+    /// Copy Constructor
+    GeometryContainer(GeometryContainer const& rOther)
+        : mGeometries(rOther.mGeometries)
+    {}
+
+    /// Components Constructor
+    GeometryContainer(
+        typename GeometriesContainerType::Pointer NewGeometries)
+        : mGeometries(NewGeometries)
+    {}
+
+    /// Destructor
+    ~GeometryContainer() = default;
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    GeometryContainer Clone()
+    {
+        typename GeometriesContainerType::Pointer p_geometries(new GeometriesContainerType(*mGeometries));
+
+        return GeometryContainer(p_geometries);
+    }
+
+    void Clear()
+    {
+        mGeometries.clear();
+    }
+
+    ///@}
+    ///@name Geometries
+    ///@{
+
+    /// Return number of geometries stored inside this geometry container
+    SizeType NumberOfGeometries() const
+    {
+        return mGeometries.size();
+    }
+
+    ///@}
+    ///@name Add Functions
+    ///@{
+
+    /// Adds a geometry to the geometry container.
+    GeometryIterator AddGeometry(typename TGeometryType::Pointer pNewGeometry)
+    {
+        auto i = mGeometries.find(pNewGeometry.Id());
+        if(i == mGeometries.end())
+            return mGeometries.insert(pNewGeometry);
+        else
+        {
+            KRATOS_ERROR << "Geometry with Id: " << pNewGeometry->Id()
+                << " exists already.";
+        }
+    }
+
+    ///@}
+    ///@name Get Functions
+    ///@{
+
+    /// Returns the Geometry::Pointer corresponding to its Id
+    typename TGeometryType::Pointer pGetGeometry(IndexType GeometryId)
+    {
+        auto i = mGeometries.find(GeometryId);
+        KRATOS_ERROR_IF(i == mGeometries.end())
+            << " geometry index not found: " << GeometryId << ".";
+        return (i.base()->second);
+    }
+
+    /// Returns the Geometry::Pointer corresponding to its name
+    typename TGeometryType::Pointer pGetGeometry(std::string GeometryName)
+    {
+        auto hash_index = TGeometryType::GenerateId(GeometryName);
+        auto i = mGeometries.find(hash_index);
+        KRATOS_ERROR_IF(i == mGeometries.end())
+            << " geometry index not found: " << GeometryName << ".";
+        return (i.base()->second);
+    }
+
+    /// Returns a reference geometry corresponding to the id
+    TGeometryType& GetGeometry(IndexType GeometryId)
+    {
+        auto i = mGeometries.find(GeometryId);
+        KRATOS_ERROR_IF(i == mGeometries.end()) << " geometry index not found: " << GeometryId << ".";
+        return *i;
+    }
+
+    /// Returns a const reference geometry corresponding to the name
+    const TGeometryType& GetGeometry(std::string GeometryName) const

```

Comment:
There is a const of string and normal with id. You might duplicate them

Reply:
Thx, I updated it.